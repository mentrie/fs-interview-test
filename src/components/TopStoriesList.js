import React from 'react';
import PropTypes from 'prop-types';
import { List, ListItem } from 'react-mdl';
import Link from 'react-router-dom/Link';

const TopStoriesList = ({stories}) => {
  const renderListItems = () => {
    return stories.map(story => (
      <div key={story.id}>
        <Link to={`/${story.id}`}>
          <ListItem>{story.title}</ListItem>
        </Link>
      </div>))
  }

  return (
    <List>
      {renderListItems()}
    </List>
  )
}

export default TopStoriesList;


