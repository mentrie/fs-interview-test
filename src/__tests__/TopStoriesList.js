import React from 'react';
import renderer from 'react-test-renderer';
import TopStoriesList from '../components/TopStoriesList';

it('Should render component with empty props',  () => {
  // jest.mock('react-router-dom/Router', () => 'Router');  
  // jest.mock('react-router-dom/Link', () => 'Link');

  const component = renderer.create(<TopStoriesList stories={[]} />);

  expect(component.toJSON()).toMatchSnapshot();
})

it('Should render component with props provided', () => {
  // jest.mock('react-router-dom/Router', () => 'Router');
  // jest.mock('react-router-dom/Link', () => 'Link');
  
  
  const component = renderer.create(<TopStoriesList stories={[{title: 'Testing', id: 1}, {title: 'testing 2', id: 2}]} />);
  expect(component.toJSON()).toMatchSnapshot();
});

