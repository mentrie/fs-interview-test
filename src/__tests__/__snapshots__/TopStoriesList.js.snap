// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component with empty props 1`] = `
<ul
  className="mdl-list"
/>
`;
